### ISSUE TITLE
Dependabot doesn't update major versions

### ISSUE THREAD

**Issue by just-jeb** [dependabot/feedback#855]
Dependencies in my [repo](https://github.com/just-jeb/angular-builders) are not getting updated with new major versions.  

Example: [`jest: ^24.7.1`](https://github.com/just-jeb/angular-builders/blob/master/package.json#L40), latest is `25`, no PR created by dependabot.  

I think it started when I removed the lockfiles (I don't use them anymore) but I can't be sure.

**Comment by feelepxyz** [comment]
@just-jeb looks like you need to manually update both `ts-jest` and `jest` as `ts-jest` has a strict peer dependency on the major version of jest. Dependabot's logic for grouped updates with peer dependency is not perfect so it's failing in this case.

I'd love to improve this logic in future but we're currently busing moving Dependabot Preview functionality into the GitHub native version of Dependabot.

**Comment by just-jeb** [comment]
Got you. Thanks for the swift response!

**Comment by jurre** [comment]
Hi @just-jeb, looking at the output when running npm on your project when trying to update typescript to 3.8, it looks like it has a peer dependency on angular:

```
npm WARN @angular/compiler-cli@9.0.5 requires a peer of typescript@>=3.6 <3.8 but none is installed. You must install peer dependencies yourself.
```

**Comment by just-jeb** [comment]
So this is what prevents from dependabot to update the version? In this case shouldn't it create a failing PR?

**Comment by feelepxyz** [comment]
> So this is what prevents from dependabot to update the version? In this case shouldn't it create a failing PR?

We've chosen not to open PRs for updates that might be broken as it might not be caught by CI and would have to introduce another update type that automerge should ignore. We're planning to make it easier to see what Dependabot couldn't update by exposing this in the GitHub Dependency Graph.

**Comment by just-jeb** [comment]
So currently there is no way to know that the dependency cannot be updated other than just keeping an eye on the repo?

**Comment by feelepxyz** [comment]
> So currently there is no way to know that the dependency cannot be updated other than just keeping an eye on the repo?

Yeah, the only way currently is looking at the update logs. To know for sure you basically have to run a full dry-run using dependabot-core and step through the code as it varies a lot by package manager.

**Comment by just-jeb** [comment]
@feelepxyz That's a bummer. Do you have any plans for tackling this issue in an automatic way (like raising a notification if the update process encountered errors)? 

Today, if you have a dependencies update problem it's completely hidden from you. The only way to know there is a problem is to notice yourself that there is a new version of some package and it didn't get updated in the repo.  
Very hard to keep track on those, especially if you have multiple repos.   

It's kind of disrupts the whole `auto-update` thing, which essentially has two main goals: 
1. Keep your dependencies up to date
2. Notify you if something is wrong (if it can't keep your dependencies up to date).  

Don't you think so?

**Comment by feelepxyz** [comment]
@just-jeb yep definitely want to do something about it but probably going to highlight this in the GitHub UI and maybe allow you to set some kind of policy there to alert about outdated dependencies, which is similar to notifying about unmaintained dependencies (something we've thought about in relation to security updates).

It's a tricky feature to get right as everyone will want slightly different rules around it.
